internal/cli/runner_install.go

```diff
@@ -0,0 +1,304 @@
+			platform := runnerinstall.Platforms[name]
+			platformSet := set.NewSet(name + " Options")
+			platform.InstallFlags(platformSet)
+		}
+	})
+}
+
+func (c *RunnerInstallCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint runner install [options]
+
+  Install a Waypoint runner to an existing platform. The platform should be
+  specified as kubernetes, nomad, ecs, or docker.
+
+  By default, this will adopt the runner after it is installed. The install will
+  attempt to install a runner for the server configured in the current Waypoint 
+  context.
+
+` + c.Flags().Help())
+}
+
+func (c *RunnerInstallCommand) Synopsis() string {
+	return "Install a Waypoint runner to Kubernetes, Nomad, ECS, or Docker"
+}
+
+func (c *RunnerInstallCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	ctx := c.Ctx
+	log := c.Log.Named("install")
+	defer c.Close()
+
+	if len(c.platform) == 0 {
+		c.ui.Output(
+			"The -platform flag is required.",
+			terminal.WithErrorStyle(),
+		)
+		c.ui.Output(c.Flags().Help(), terminal.WithInfoStyle())
+		return 1
+	}
+	p, ok := runnerinstall.Platforms[strings.ToLower(c.platform[0])]
+	if !ok {
+		c.ui.Output(
+			"Error installing server into %q: unsupported platform",
+			c.platform,
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	sg := c.ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Connecting to: %s", c.serverUrl)
+	defer func() { s.Abort() }()
+
+	if c.adopt && c.serverCookie == "" {
+		serverConfig, err := c.project.Client().GetServerConfig(ctx, &empty.Empty{})
+		if err != nil {
+			c.ui.Output(
+				"Error getting server config.",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+		c.serverCookie = serverConfig.Config.Cookie
+	} else if c.adopt && c.serverCookie != "" {
+		serverConfig, err := c.project.Client().GetServerConfig(ctx, &empty.Empty{})
+		if err != nil {
+			c.ui.Output(
+				"Error getting server config.",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+		if c.serverCookie != serverConfig.Config.Cookie {
+			c.ui.Output(
+				"Incorrect server cookie provided.",
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+	}
+
+	if c.serverUrl == "" {
+		c.ui.Output(
+			"-server-addr must be supplied for adoption.",
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	client := c.project.Client()
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	// We generate the ID if the user doesn't provide one
+	// This ID is used later to adopt the runner
+	var id string
+	var err error
+	if c.id == "" {
+		id, err = server.Id()
+		if err != nil {
+			c.ui.Output("Error generating runner ID: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+	} else {
+		id = c.id
+	}
```

Comment:
It's been a while since I fought with this! But I did try out this suggestion locally and the compiler actually thinks that `id` on line 222 in your suggestion is a new, unused variable `id` 😅 even though it is indeed used later on `p.Install`'s options.

```terminal-output
# github.com/hashicorp/waypoint/internal/cli
internal/cli/runner_install.go:199:3: id declared but not used
make: *** [bin] Error 2
```

But if I partially implement your suggestion, and still declare `err` as a var of type `error` before the if block, I don't need to use `:=` in the if block, and the compiler is happy.
```suggestion
	id := c.id
	var err error
	if id == "" {
		id, err = server.Id()
		if err != nil {
			c.ui.Output("Error generating runner ID: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
			return 1
		}
	} 
```

Reply:
Sounds good! Def trust the compiler over me 😄